### PR TITLE
feat(ui+api): Move "Account Subscriptions" to React

### PR DESCRIPTION
**Pull request by billyvg**
image

### CODE REVIEW

**Comment by getsentry-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">You should update CHANGES due to the size of this PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

File: src/sentry/api/endpoints/account_subscriptions.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        :param int list_id: id of newsletter list
+        :param boolean subscribed: should be subscribed to newsletter
+        :auth: required
+        """
+
+        user = request.user
+        email = UserEmail.get_primary_email(user)
+        subscribed = request.DATA.get('subscribed') == '1'
+        try:
+            list_id = int(request.DATA.get('list_id', ''))
```
Comment (dcramer) [inline src/sentry/api/endpoints/account_subscriptions.py]: nit: probably would be good to keep this camelCase for consistency (listId, or just list)

**Comment by MaxBittker** [comment]
i think if we're duplicating some of these forms into endpoints, it's good to write some lightweight tests for them, so we have something when the forms are eventually deleted

File: src/sentry/api/endpoints/user_subscriptions.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        :param int listId: id of newsletter list
+        :param boolean subscribed: should be subscribed to newsletter
+        :auth: required
+        """
+
+        email = UserEmail.get_primary_email(user)
+
+        # Can't handle subscriptions without a verified email
+        if not email.is_verified:
+            return Response({'details': 'Verify your email address first.'},
```
Comment (MaxBittker) [inline src/sentry/api/endpoints/user_subscriptions.py]: this response isn't super good if it's displayed clientside, something more declarative maybe? "must have verified email to subscribe to newsletter"?

**Review by MaxBittker — APPROVED** [review]
lgmt, sorry for the slow review

**Comment by getsentry-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">You should update CHANGES due to the size of this PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

**Pull request merged**